### PR TITLE
Disable create snapshot button if not assigned to an application or instance

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Follow up to https://github.com/FlowFuse/flowfuse/pull/3026

Screenshot 2023-11-17 at 15 06 26

## Related Issue(s)

N/a

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/device/DeveloperMode/index.vue [hunk 1]
```diff
@@ -42,14 +42,19 @@
...
+                                :disabled="createSnapshotDisabled"
+                                kind="secondary"
+                                class="w-28 whitespace-nowrap"
+                                size="small"
+                                data-action="create-snapshot-dialog"
+                                @click="showCreateSnapshotDialog"
+                            >
+                                Create Snapshot
+                            </ff-button>
+                            <span v-if="createSnapshotDisabled" class="ff-description ml-2">A device must first be assigned to an Application, in order to create snapshots.</span>
```
Comment (knolleary) [inline frontend/src/pages/device/DeveloperMode/index.vue:56]:
```suggestion
                            <span v-if="createSnapshotDisabled" class="ff-description ml-2">A device must first be assigned to an Application or Instance in order to create snapshots.</span>
```

**Review by knolleary — APPROVED** [review]
Approved, pending a slight wording change proposed

**Comment by knolleary** [comment]
Actually, want to dbl check something again...

**Comment by knolleary** [comment]
The Create Snapshot button should only be disabled if the device is not assigned to anything.

If assigned to an Application **or Instance** - it should be enabled.

**Comment by knolleary** [comment]
I missed this in #3026 as well - so I think there are some bits to redo from that PR. My apologies for missing it.

**Comment by knolleary** [comment]
I'm going in loops and confusing myself over what should be possible and when. Sorry for the multiple comments.

In Developer Mode, the Create Snapshot button should be enabled when assigned to *either* Application or Instance.

The Snapshots tab in the Device view disappears when Device is assigned to an Instance. That is a choice... which I'm not revisiting here.

File: frontend/src/pages/device/DeveloperMode/index.vue [hunk 2]
```diff
@@ -110,6 +115,9 @@ export default {
         },
         editorCanBeEnabled: function () {
             return this.developerMode && this.device.status === 'running'
+        },
+        createSnapshotDisabled () {
+            return this.device.ownerType !== 'application'
```
Comment (knolleary) [inline frontend/src/pages/device/DeveloperMode/index.vue:120]: To summarise the barrage of contradictory comments yesterday, a change is needed here - on the Developer Mode view, the Create Snapshot button should enabled if the device is assigned to either an instance or application.

**Review by knolleary — CHANGES_REQUESTED** [review]

File: frontend/src/pages/device/DeveloperMode/index.vue (hunk 2 above)
Comment (Pezmc) [reply frontend/src/pages/device/DeveloperMode/index.vue]: [37072cc](https://github.com/FlowFuse/flowfuse/pull/3049/commits/37072ccac45f51fade968740a892aaad88a25683)

**Review by knolleary — APPROVED** [review]

**Pull request merged**